### PR TITLE
Feature: Room adverts

### PR DESCRIPTION
**Pull request by Ellen-Wittingen**
Added room adverts. They are accessible via [http://localhost:4200/room-adverts](http://localhost:4200/room-adverts) and via the menu-item in the left sidebar.

There is also a view for unauthorized users on the public part of the website, it is accessible via [http://localhost:4200/public/room-forum](http://localhost:4200/public/room-forum) and via the menu-item "Kamerforum" in the "Over ons" dropdown.

Please also check if the English translations are proper English, as nobody except from me has looked at them so far.

Note: it requires the following amber-api branch to function properly: [https://github.com/csvalpha/amber-api/pull/393](https://github.com/csvalpha/amber-api/pull/393) NOTE BY MATTEO: this backend branch has already been incorporated into staging

### CODE REVIEW

**Review by DrumsnChocolate — CHANGES_REQUESTED** [review]
The implementation works as expected, but I see you haven't quite used ember octane. I know that lots of the entire app is still written in the old style, but please try to follow the checklist (most important): https://guides.emberjs.com/release/upgrading/current-edition/glimmer-components/
and if relevant the rest of the octane upgrade guide mentioned there.
For example, named arguments means we should be using `@roomAdvert.location` in our templates, not `roomAdvert.location`
And we should not be using `import Component from '@ember/component';` in any of our new components. Always use native classes for everything, and for components only ever extend glimmer components.

File: translations/en.yaml
Comment (DrumsnChocolate) [inline translations/en.yaml:305]:
```suggestion
    roomForum: Room forum
```

**Review by DrumsnChocolate — CHANGES_REQUESTED** [review]

File: app/components/header-nav.js
Comment (guidojw) [inline app/components/header-nav.js:7]: These are legacy Ember component API and Glimmer won't do anything with them. You need to wrap the stuff in the template in these accordingly: https://guides.emberjs.com/release/upgrading/current-edition/glimmer-components/#toc_outer-html

**Review by guidojw — CHANGES_REQUESTED** [review]

File: app/components/header-nav.js
Comment (guidojw) [inline app/components/header-nav.js:19]: What's up with this?

**Comment by DrumsnChocolate** [comment]
The text can become rather long, and the text is not vertically bounded in display:
afbeelding

**Comment by DrumsnChocolate** [comment]
afbeelding
When logged in, I don't see any shorttext preview of the written content of the advert. Is this something we should add?

**Comment by DrumsnChocolate** [comment]
These are the main UX points I found. I will leave a code review later, this Friday evening probably

**Comment by Ellen-Wittingen** [comment]
Fixed the long scrolling content. Now there is a "lees meer" button that opens a modal with a scrollable bigger view of the advert when clicked upon. That modal is also opened when clicking anywhere on the advert.

image

image

**Comment by Ellen-Wittingen** [comment]
Fixed not showing a short description on room advert index page:

image

**Comment by Ellen-Wittingen** [comment]
Apparently I never pushed the fixes I mentioned, oops

**Comment by DrumsnChocolate** [comment]
We're reviewing this once more, @lodewiges and @Notheen01 will you two also remember to leave a review before we meet on Monday?

@Ellen-Wittingen we will finally be merging this thing :)

**Comment by DrumsnChocolate** [comment]
afbeelding
icon doesn't load for some reason. Unsure if that's my fault

**Comment by DrumsnChocolate** [comment]
yes it was my fault: I did `yarn install` in the staging branch. `yarn install` in the `feature/rooms` resolved the icon issue :facepalm:

**Comment by DrumsnChocolate** [comment]
Would it be practical to have a subdomain like `kamers.csvalpha.nl` point to `csvalpha.nl/public/room-forum` ? That would allow for more external traffic, because it is easier to share and remember.

**Review by DrumsnChocolate — CHANGES_REQUESTED** [review]
Overall, nice work! Very happy that you adhere well to octane (as far as I can see).

Thanks Ellen! My comments in this review may not be exhaustive, so let's wait for @lodewiges and @Notheen01 reviews. The three of us will discuss them on Monday, and I'll let you know when we think this PR is ready for merging.

File: app/components/header-nav.hbs
Comment (lodewiges) [inline app/components/header-nav.hbs:1]: small question: what is the advantage of adding this. for a lot of items

File: app/components/header-nav.js
Comment (lodewiges) [inline app/components/header-nav.js:64]: what is this checking for?

File: app/components/public/index/header.js
Comment (lodewiges) [inline app/components/public/index/header.js:1]: why is this ember component and not glitter component

File: app/templates/public/room-forum.hbs
Comment (lodewiges) [reply app/templates/public/room-forum.hbs:42]: could you explain the source of this bug Monday?

**Review by lodewiges — COMMENTED** [review]
Read through it and the code looks mostly good had one comment but in the end I saw lint made the same comment

**Comment by lodewiges** [comment]
> Would it be practical to have a subdomain like `kamers.csvalpha.nl` point to `csvalpha.nl/public/room-forum` ? That would allow for more external traffic, because it is easier to share and remember.

sounds good lets discuss the advantages Monday

**Comment by lodewiges** [comment]
staging has been updated in the meantime so we need to merge it again

**Comment by lodewiges** [comment]
I am not sure if we can change this but the house name box gets recognized as address field instead of the address field itself. changing this would make it easier to fill in a advert.
image

**Comment by lodewiges** [comment]
it think it would be better if the date could be select through a small calender
image

**Comment by lodewiges** [comment]
when on the external website going to the kamerforum page, scrolling down opening a post and closing this post again.
you're unable to scroll

**Comment by lodewiges** [comment]
Why did we hide the kamerforum below over ons instead of giving it its own place in the header, I thing this would be more appropriate cause people see it soon instead of needing to look in a drop down menu

File: app/templates/public/room-forum.hbs
Comment (lodewiges) [reply app/templates/public/room-forum.hbs:42]: How are you able to have the sidebar and be on the external website UI?

**Comment by lodewiges** [comment]
overall looks good, some small things, style of the page is nicely inline with the rest of the site. Nice work Ellen!

**Comment by DrumsnChocolate** [comment]
>it think it would be better if the date could be select through a small calender instead off text.  This also should be a required field in my opinion

I am inclined to agree

> when on the external website going to the kamerforum page, scrolling down opening a post and closing this post again.
you're unable to scroll

unable to reproduce

> Why did we hide the kamerforum below over ons instead of giving it its own place in the header, I thing this would be more appropriate cause people see it soon instead of needing to look in a drop down menu

take a look at the designs supplied by promocie. if you disagree, let's first get this merged and then have a talk to the promocie.

File: app/templates/public/room-forum.hbs
Comment (DrumsnChocolate) [reply app/templates/public/room-forum.hbs]: fixed

File: app/components/header-nav.js
Comment (DrumsnChocolate) [reply app/components/header-nav.js:64]: discussed in person

File: app/components/public/index/header.js
Comment (DrumsnChocolate) [inline app/components/public/index/header.js:7]: @Ellen-Wittingen this is not used anywhere. Was this functionality finished? I presume it is meant to randomly select from a bunch of 'kamer-forum banners'?

**Comment by DrumsnChocolate** [comment]
> I am not sure if we can change this but the house name box gets recognized as address field instead of the address field itself. changing this would make it easier to fill in a advert.

what browser was this again? then I can check if this happens to me too


regarding the input box for available from, I'm working on that: https://github.com/csvalpha/amber-api/pull/439
I will add changes here in ui still.

**Comment by lodewiges** [comment]
it was in Microsoft edge

**Comment by Ellen-Wittingen** [comment]
So everything is good, except that the "available from" field will be changed to a date instead of text by @DrumsnChocolate ? So do I still have to do anything or will you guys merge it when you are done?

File: app/components/forms/room-advert-form.hbs
Comment (Ellen-Wittingen) [reply app/components/forms/room-advert-form.hbs:6]: Oh, I did not see this. I just changed one back to <form {{action @onSubmit on='submit'}}> because it did not work. See my last commit

**Comment by DrumsnChocolate** [comment]
> So everything is good, except that the "available from" field will be changed to a date instead of text by @DrumsnChocolate ? So do I still have to do anything or will you guys merge it when you are done?

I'll merge it when I've taken care of the remaining two points: autofill address field, and the type change of the availableFrom field.

**Comment by DrumsnChocolate** [comment]
@Ellen-Wittingen I do not have any issues with the form no longer submitting when I reapply my change. Are you sure my change broke things?

**Comment by DrumsnChocolate** [comment]
what am I doing lmao, that's a completely different form

**Comment by DrumsnChocolate** [comment]
depends on https://github.com/csvalpha/amber-api/pull/439

**Review by DrumsnChocolate — APPROVED** [review]

**Review by lodewiges — APPROVED** [review]

**Pull request merged**
